src/electionguard/discrete_log.py

```diff
@@ -6,16 +6,39 @@
 
 from .constants import get_generator
 from .singleton import Singleton
-from .group import ElementModP, ONE_MOD_P, mult_p
+from .group import BaseElement, ElementModP, ONE_MOD_P, mult_p
 
-DLogCache = Dict[ElementModP, int]
-DLOG_MAX = 100_000_000
-"""The max number to calculate.  This value is used to stop a race condition."""
+DiscreteLogCache = Dict[ElementModP, int]
+
+_DLOG_MAX_EXPONENT = 100_000_000
+"""The max exponent to calculate.  This value is used to stop a race condition."""
+
+_INITIAL_CACHE = {ONE_MOD_P: 0}
+
+
+class DiscreteLogExponentError(ValueError):
+    """Raised when the max exponent is larger than the system allows."""
+
+    def __init__(self, exponent: int, max_exponent: int = _DLOG_MAX_EXPONENT) -> None:
+        message = (
+            f"Discrete log exponent of {exponent} exceeds maximum of {max_exponent}."
+        )
+        super().__init__(message)
```

Comment:
The next error has the message directly in the init call.  We should be consistent and make the two match.